### DISCUSSION TITLE
Propagate parsed path params to `validateSearch`

### DISCUSSION THREAD

**Discussion by mversic** [TanStack/router#2927] [Ideas]
I require to know what exact URL am I validating **search** parameters for.

### Example
Imagine I have a URL defined as `/booking/$pickupDate?returnDate=2025-03-21`.
I want to validate that `returnDate` **search** param is before `pickupDate` **path** param

### Solution
I suggest to modify `validateSearch(searchParams)` to `validateSearch(searchParams, pathParams)`

**Comment by mversic** [comment]
any comments on this?

**Comment by 5ebastianMeier** [comment]
I think this is out of scope of the searchParams validation, because it is not a schema validation based on searchParams alone.
You could use the `beforeLoad` callback of the route that carries the `pickupDate` param. The `context` argument of this callback gives you access to the searchParams as well as the params so if your validation fails you could redirect to an error page.

**Comment by mversic** [comment]
thank you, it works as you described. The downside is that I cannot modify `search` params or provide a fallback inside `beforeLoad` as I can in `validateSearch`

**Reply by 5ebastianMeier** [reply]
Depending on what actions you want to perform based on your validation result you might be able to modify searchParams or navigate to a fallback page by using a `redirect`. If a `redirect` doesn't work for you please explain what you're struggling with and example code if possible.

**Reply by mversic** [reply]
Sounds good. I have no struggles
